### PR TITLE
Example for ImageBuilder build pipeline for amazon linux 2023

### PR DESCRIPTION
**Pull request by hamadsuniverse**
Added an example project that demonstrates how to create an ImageBuilder pipeline to build a container image using Amazon Linux 2023 base image and installing git, docker and nodejs in it so it can be used in building pipelines such as in Amazon CodeCatalyst.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: typescript/imagebuilder/bin/imagebuilder-components/docker.yaml
Comment (kaizio) [inline typescript/imagebuilder/bin/imagebuilder-components/docker.yaml:1]: This file is installing docker not git. Check names used in the file.

File: typescript/imagebuilder/lib/imagebuilder-stack.ts
Comment (kaizio) [inline typescript/imagebuilder/lib/imagebuilder-stack.ts:1]: Can you add comments for this file so that the reader knows what is happening.

**Review by kaizio — CHANGES_REQUESTED** [review]
Requested a couple of changes. Thanks for the add.

**Comment by kaizio** [comment]
Also `npm run --if-present test` is failing in the build checks.

**Comment by hamadsuniverse** [comment]
Actioned all points and build/all checks are passing. Pending your review and merge. Thanks!

**Review by kaizio — APPROVED** [review]

**Pull request merged**